lib/src/chart/bar_chart/bar_chart_painter.dart

```diff
@@ -226,7 +226,20 @@ class BarChartPainter extends AxisChartPainter<BarChartData> with TouchHandler<B
                 bottomRight: borderRadius.bottomRight);
           }
 
-          _barPaint.color = barRod.backDrawRodData.color;
+          if (barRod.colors.length > 1) {
+            final rect = Rect.fromLTRB(
+              barRRect.left,
+              barRRect.top,
+              barRRect.right,
+              barRRect.bottom,
+            );
+            _barPaint.shader = LinearGradient(colors: [
+              barRod.backDrawRodData.color,
+              barRod.backDrawRodData.color,
+            ]).createShader(rect);
+          } else
+            _barPaint.color = barRod.backDrawRodData.color;

```

Comment:
And it is an unwritten code style rule, but please put { } around the single line statement too.